Flutter/datagrid/filtering.md

```diff
@@ -0,0 +1,616 @@
+              'ID',
+              FilterCondition(
+                value: 1005,
+                type: FilterType.contains,
+                filterBehavior: FilterBehavior.stringDataType,
+              ),
+            );
+          }),
+    ],
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Filter operator
+
+The `FilterOperator` enum property is used to decide how a logical operator is to be applied between multiple filter conditions.
+
+* **and** - `AND` logical operator applies between multiple filter conditions 
+* **or** - `OR` logical operator applies between multiple filter conditions
+
+{% tabs %}
+{% highlight Dart %} 
+
+@override
+Widget build(BuildContext context) {
+  return Column(
+    children: [
+      Expanded(
+        child: SfDataGrid(source: _employeeDataSource, columns: [
+          GridColumn(
+              columnName: 'ID',
+              label: Container(
+                  padding: EdgeInsets.symmetric(horizontal: 16.0),
+                  alignment: Alignment.centerRight,
+                  child: Text(
+                    'ID',
+                    overflow: TextOverflow.ellipsis,
+                  ))),
+          GridColumn(
+              columnName: 'Name',
+              label: Container(
+                  padding: EdgeInsets.symmetric(horizontal: 16.0),
+                  alignment: Alignment.centerLeft,
+                  child: Text(
+                    'Name',
+                    overflow: TextOverflow.ellipsis,
+                  ))),
+          GridColumn(
+              columnName: 'Designation',
+              label: Container(
+                  padding: EdgeInsets.symmetric(horizontal: 16.0),
+                  alignment: Alignment.centerLeft,
+                  child: Text(
+                    'Designation',
+                    overflow: TextOverflow.ellipsis,
+                  ))),
+          GridColumn(
+              columnName: 'Salary',
+              label: Container(
+                  padding: EdgeInsets.symmetric(horizontal: 16.0),
+                  alignment: Alignment.centerRight,
+                  child: Text(
+                    'Salary',
+                    overflow: TextOverflow.ellipsis,
+                  ))),
+        ]),
+      ),
+      MaterialButton(
+          child: Text('Add Filter'),
+          onPressed: () {
+            _employeeDataSource.addFilter(
+              'ID',
+              FilterCondition(
+                value: 1005,
+                filterOperator: FilterOperator.and,
+                type: FilterType.greaterThanOrEqual,
+              ),
+            );
+
+            _employeeDataSource.addFilter(
+              'ID',
+              FilterCondition(
+                value: 1010,
+                filterOperator: FilterOperator.and,
+                type: FilterType.lessThanOrEqual,
+              ),
+            );
+          }),
+    ],
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## UI Filtering

```

Comment:
@ashok-kuvaraja  - Add the following topic above to this topic,
**Retrieving the filtered rows**
https://help.syncfusion.com/windowsforms/datagrid/filtering#retrieving-the-filtered-records

Then, add below topic,

**Filter DateColumn with range between two dates**
https://help.syncfusion.com/windowsforms/datagrid/filtering#filter-datecolumn-with-range-between-two-dates

Reply:
Included the topics.